tests/components/sensor/test_filter.py

```diff
@@ -33,7 +39,10 @@ def test_setup_fail(self):
 
     def test_chain(self):
         """Test if filter chaining works."""
+        self.init_recorder()
```

Comment:
This doesn't test anything. I would expect a new test that mocks `history.state_changes_during_period` to return a list of state objects that would then influence the filter data after setup.

Reply:
I wasn't really testing the history feature 😅 it was just acknowledging the dependency

Will workout a proper test 👍